### ISSUE TITLE
Crash on passing closure to `options` of `MultiSelect` component

### ISSUE THREAD

**Issue by CaptainIRS** [filamentphp/filament#4222]
### Package

filament/forms

### Package Version

v2.16.17

### Laravel Version

v9.31.0

### Livewire Version

v2.10.7

### PHP Version

8.1.10

### Problem description

When a closure is passed to the `options` method of `MultiSelect` component, a JS runtime error is thrown when the multi-select dropdown is clicked in the browser:

```
module.esm.js?t=1664257141009:28000 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'includes')
    at module.esm.js?t=1664257141009:28000:65
    at Array.map (<anonymous>)
    at Proxy.getOptions (module.esm.js?t=1664257141009:28000:37)
    at async Proxy.getChoices (module.esm.js?t=1664257141009:27982:33)
    at async Proxy.refreshChoices (module.esm.js?t=1664257141009:27976:25)
    at async HTMLSelectElement.<anonymous> (module.esm.js?t=1664257141009:27934:13)
```

I have bisected the issue and this bug starts to occur from v2.16.16 (doesn't occur in v2.16.15 and earlier). I suspect this was introduced in #4195.

### Expected behavior

The options that are returned by the closure are displayed correctly.

### Steps to reproduce

* Create a MultiSelect component and pass a closure to the options method.
```php
protected function getFormSchema(): array
    {
        return [
            MultiSelect::make('test')
                ->options(fn () => [
                    'test1' => 'Test 1',
                    'test2' => 'Test 2',
                    'test3' => 'Test 3',
                ])
        ];
    }
```
* Open the form in the browser and click the multi-select field
* A runtime error is thrown in the console and no options are shown in the dropdown.

![image](https://user-images.githubusercontent.com/36656347/192443966-57adae74-e2ce-410a-b10d-9d56f245ec4b.png)

### Reproduction repository

https://github.com/CaptainIRS/filamentphp-bug

**Comment by danharrin** [comment]
I definitely tested this use case, as the PR I made fixed the issues with dynamic multiselect options.

I think that this error is caused by you not calling `$this->form->fill()` from `mount()`. It is mandatory for every form.

**Comment by danharrin** [comment]
I just tested again, I can confirm that following the docs with `$this->form->fill()` fixes the problem:

```php
public function mount()
{
    $this->form->fill();
}
```

**Comment by CaptainIRS** [comment]
I did correctly initialise the form in the actual codebase, but sorry, I missed that when creating the repro. I think the actual issue is that `MultiSelect` no longer takes `null` as a valid value after v2.16.16. I was trying to follow https://youtu.be/W_eNyimRi3w?t=204 where `$set('postId', null)` is done to clear the value of the `Select` component. I did the same with the `MultiSelect` component, which used to clear the value correctly until v2.16.15, but after the version it throws the error described in the issue. So, now the only way to clear the value is to do `$set('postId', [])`.
